patches/chrome-installer-mini_installer-mini_installer.cc.patch

```diff
@@ -11,7 +11,133 @@ index 1f39abbeb2579098a54fa82612199f0307ea03b4..be08f89c43599fc766048d230572e71a
  // Opens the Google Update ClientState key. If |binaries| is false, opens the
  // key for Google Chrome or Chrome SxS (canary). If |binaries| is true and an
  // existing multi-install Chrome is being updated, opens the key for the
-@@ -873,7 +873,7 @@ ProcessExitResult WMain(HMODULE module) {
+@@ -142,6 +142,105 @@ void SetInstallerFlags(const Configuration& configuration) {
+ }
+ #endif  // GOOGLE_CHROME_BUILD
+ 
++#if defined(BRAVE_CHROMIUM_BUILD)
++typedef StackString<128> ReferralCodeString;
++
++bool ParseStandardReferralCode(const wchar_t* filename, ReferralCodeString* referral_code) {
++  // Scan backwards for last dash in filename.
```

Comment:
This approach definitely works- but curious why you wouldn't use `std::wstring` (avoiding a loop); ex:
```
std::wstring filename2 = std::wstring(filename);
std::wstring part = filename2.substr(filename2.find_last_of("-"));
if (part.length() != 6) return false;
```

Reply:
No, that's a really good question. The mini_installer is kept really tiny and it specifically doesn't include STL or CRT calls. The header block at the top of mini_installer.cc has a really good explanation of this:

```
// mini_installer.exe is the first exe that is run when chrome is being
// installed or upgraded. It is designed to be extremely small (~5KB with no
// extra resources linked) and it has two main jobs:
//   1) unpack the resources (possibly decompressing some)
//   2) run the real installer (setup.exe) with appropriate flags.
//
// In order to be really small the app doesn't link against the CRT and
// defines the following compiler/linker flags:
//   EnableIntrinsicFunctions="true" compiler: /Oi
//   BasicRuntimeChecks="0"
//   BufferSecurityCheck="false" compiler: /GS-
//   EntryPointSymbol="MainEntryPoint" linker: /ENTRY
//   IgnoreAllDefaultLibraries="true" linker: /NODEFAULTLIB
//   OptimizeForWindows98="1" linker: /OPT:NOWIN98
//   linker: /SAFESEH:NO
```

Reply:
That makes perfect sense, thanks 😄